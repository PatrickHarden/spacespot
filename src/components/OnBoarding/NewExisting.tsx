import React from 'react'
import { useIntl } from 'react-intl'
import styled from 'styled-components'
import OutlineButton, { ButtonProps } from './OutlineButton'
import Heading4 from 'components/common/Heading4'

const Container = styled.div`
  padding: 0%;
  margin-bottom: 50px;
`
const ButtonContainer = styled.div`
  display: flex;
  flex: 1;
`

const ButtonOnboarding = styled(OutlineButton)<ButtonProps>`
  text-align: center;
  padding: 28px 10px;
  margin-right: 20px;
  overflow: visible;
  :hover { ${props =>
    props.selected || props.disabled === true
      ? ''
      : `border: 1px solid #6cb9d5;
         box-sizing: border-box;
         box-shadow: 0 1px 4px 0 #6cb9d5;
         cursor: pointer`}
`

export interface NewExistingProps {
  onClick: (type: boolean) => void
  selected: boolean | undefined
  hasBuildings: boolean
}

const NewExisting = (props: NewExistingProps) => {
  const { onClick, selected, hasBuildings } = props
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })

  return (
    <Container>
      <Heading4>{t('ONBOARDING_NEW_EXISTING')}</Heading4>
      <ButtonContainer>
        <ButtonOnboarding
          testId="existing-building"
          disabled={!hasBuildings}
          onClick={() => onClick(false)}
          selected={selected === false}>
          {t('ONBOARDING_EXISTING')}
        </ButtonOnboarding>
        <ButtonOnboarding
          testId="new-building"
          onClick={() => onClick(true)}
          selected={selected === true}>
          {t('ONBOARDING_NEW_BUILDING')}
        </ButtonOnboarding>
      </ButtonContainer>
    </Container>
  )
}
export default NewExisting
