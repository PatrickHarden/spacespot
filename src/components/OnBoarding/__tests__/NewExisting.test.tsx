import React from 'react'
import NewExisting from '../NewExisting'

import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

it('renders without crashing ', async () => {
  const setIsNew = jest.fn()
  const wrapper = mount(
    <NewExisting hasBuildings={true} selected onClick={setIsNew} />,
  )
  expect(mountToJson(wrapper)).toMatchSnapshot()
  const buttons = wrapper.find('button')
  act(() => {
    buttons.at(0).simulate('click')
  })
  expect(setIsNew).toHaveBeenCalledWith(false)
  act(() => {
    buttons.at(1).simulate('click')
  })
  expect(setIsNew).toHaveBeenCalledWith(true)
})

it('renders without crashing without buildings ', async () => {
  const setIsNew = jest.fn()
  const wrapper = mount(
    <NewExisting hasBuildings={false} selected onClick={setIsNew} />,
  )
  expect(mountToJson(wrapper)).toMatchSnapshot()
})
