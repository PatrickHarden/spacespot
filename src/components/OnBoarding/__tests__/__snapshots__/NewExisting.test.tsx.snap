// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing  1`] = `
<NewExisting
  hasBuildings={true}
  onClick={[MockFunction]}
  selected={true}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "lastClassName": "heZAUk",
            "rules": Array [
              "
  padding: 0%;
  margin-bottom: 50px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bxivhb heZAUk"
      >
        <styled.h4>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "ezOBiL",
                  "rules": Array [
                    "
  color: #000000;
  font-size: 18px;
  font-weight: 500;
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
",
                  ],
                },
                "displayName": "styled.h4",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "h4",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h4
              className="sc-htpNat ezOBiL"
            >
              ONBOARDING_NEW_EXISTING
            </h4>
          </StyledComponent>
        </styled.h4>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "lastClassName": "jeKXEu",
                  "rules": Array [
                    "
  display: flex;
  flex: 1;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-ifAKCX jeKXEu"
            >
              <Styled(Styled(Component))
                disabled={false}
                onClick={[Function]}
                selected={false}
                testId="existing-building"
              >
                <StyledComponent
                  disabled={false}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-EHOje",
                        "isStatic": false,
                        "lastClassName": "gATSaV",
                        "rules": Array [
                          "
  box-sizing: border-box;
  outline: none;
  border: ",
                          [Function],
                          ";
  border-radius: 2px;
  position: relative;
  background-color: ",
                          [Function],
                          ";
  min-width: 120px;
  color: #000000;
  opacity: ",
                          [Function],
                          ";
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
",
                          "
  text-align: center;
  padding: 28px 10px;
  margin-right: 20px;
  overflow: visible;
  :hover { ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "Styled(Styled(Component))",
                      "foldedComponentIds": Array [
                        "sc-bwzfXH",
                      ],
                      "render": [Function],
                      "styledComponentId": "sc-EHOje",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                  selected={false}
                  testId="existing-building"
                >
                  <Component
                    className="sc-bwzfXH sc-EHOje jPutGp"
                    disabled={false}
                    onClick={[Function]}
                    selected={false}
                    testId="existing-building"
                  >
                    <button
                      className="sc-bwzfXH sc-EHOje jPutGp"
                      data-testid="existing-building"
                      disabled={false}
                      onClick={[Function]}
                    >
                      ONBOARDING_EXISTING
                    </button>
                  </Component>
                </StyledComponent>
              </Styled(Styled(Component))>
              <Styled(Styled(Component))
                onClick={[Function]}
                selected={true}
                testId="new-building"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-EHOje",
                        "isStatic": false,
                        "lastClassName": "gATSaV",
                        "rules": Array [
                          "
  box-sizing: border-box;
  outline: none;
  border: ",
                          [Function],
                          ";
  border-radius: 2px;
  position: relative;
  background-color: ",
                          [Function],
                          ";
  min-width: 120px;
  color: #000000;
  opacity: ",
                          [Function],
                          ";
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
",
                          "
  text-align: center;
  padding: 28px 10px;
  margin-right: 20px;
  overflow: visible;
  :hover { ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "Styled(Styled(Component))",
                      "foldedComponentIds": Array [
                        "sc-bwzfXH",
                      ],
                      "render": [Function],
                      "styledComponentId": "sc-EHOje",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                  selected={true}
                  testId="new-building"
                >
                  <Component
                    className="sc-bwzfXH sc-EHOje gATSaV"
                    onClick={[Function]}
                    selected={true}
                    testId="new-building"
                  >
                    <button
                      className="sc-bwzfXH sc-EHOje gATSaV"
                      data-testid="new-building"
                      onClick={[Function]}
                    >
                      <Styled(CheckCircleIcon)
                        fontSize="default"
                        htmlColor="#6CB9D5"
                      >
                        <StyledComponent
                          fontSize="default"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "compare": null,
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bdVaJa",
                                "isStatic": false,
                                "lastClassName": "bYrkLc",
                                "rules": Array [
                                  "
  position: absolute;
  background-color: #ffffff;
  top: -11px;
  right: -11px;
  &.MuiSvgIcon-root {
    font-size: 1.5rem;
  }
",
                                ],
                              },
                              "displayName": "Styled(CheckCircleIcon)",
                              "foldedComponentIds": Array [],
                              "muiName": "SvgIcon",
                              "render": [Function],
                              "styledComponentId": "sc-bdVaJa",
                              "target": Object {
                                "$$typeof": Symbol(react.memo),
                                "compare": null,
                                "displayName": "CheckCircleIcon",
                                "muiName": "SvgIcon",
                                "type": Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "render": [Function],
                                },
                              },
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          htmlColor="#6CB9D5"
                        >
                          <ForwardRef
                            className="sc-bdVaJa bYrkLc"
                            fontSize="default"
                            htmlColor="#6CB9D5"
                          >
                            <WithStyles(ForwardRef(SvgIcon))
                              className="sc-bdVaJa bYrkLc"
                              fontSize="default"
                              htmlColor="#6CB9D5"
                            >
                              <ForwardRef(SvgIcon)
                                className="sc-bdVaJa bYrkLc"
                                classes={
                                  Object {
                                    "colorAction": "MuiSvgIcon-colorAction",
                                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                                    "colorError": "MuiSvgIcon-colorError",
                                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                    "root": "MuiSvgIcon-root",
                                  }
                                }
                                fontSize="default"
                                htmlColor="#6CB9D5"
                              >
                                <svg
                                  aria-hidden="true"
                                  className="MuiSvgIcon-root sc-bdVaJa bYrkLc"
                                  color="#6CB9D5"
                                  focusable="false"
                                  role="presentation"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                                  />
                                </svg>
                              </ForwardRef(SvgIcon)>
                            </WithStyles(ForwardRef(SvgIcon))>
                          </ForwardRef>
                        </StyledComponent>
                      </Styled(CheckCircleIcon)>
                      ONBOARDING_NEW_BUILDING
                    </button>
                  </Component>
                </StyledComponent>
              </Styled(Styled(Component))>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</NewExisting>
`;

exports[`renders without crashing without buildings  1`] = `
<NewExisting
  hasBuildings={false}
  onClick={[MockFunction]}
  selected={true}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "lastClassName": "heZAUk",
            "rules": Array [
              "
  padding: 0%;
  margin-bottom: 50px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bxivhb heZAUk"
      >
        <styled.h4>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "ezOBiL",
                  "rules": Array [
                    "
  color: #000000;
  font-size: 18px;
  font-weight: 500;
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
",
                  ],
                },
                "displayName": "styled.h4",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "h4",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h4
              className="sc-htpNat ezOBiL"
            >
              ONBOARDING_NEW_EXISTING
            </h4>
          </StyledComponent>
        </styled.h4>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "lastClassName": "jeKXEu",
                  "rules": Array [
                    "
  display: flex;
  flex: 1;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-ifAKCX jeKXEu"
            >
              <Styled(Styled(Component))
                disabled={true}
                onClick={[Function]}
                selected={false}
                testId="existing-building"
              >
                <StyledComponent
                  disabled={true}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-EHOje",
                        "isStatic": false,
                        "lastClassName": "gATSaV",
                        "rules": Array [
                          "
  box-sizing: border-box;
  outline: none;
  border: ",
                          [Function],
                          ";
  border-radius: 2px;
  position: relative;
  background-color: ",
                          [Function],
                          ";
  min-width: 120px;
  color: #000000;
  opacity: ",
                          [Function],
                          ";
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
",
                          "
  text-align: center;
  padding: 28px 10px;
  margin-right: 20px;
  overflow: visible;
  :hover { ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "Styled(Styled(Component))",
                      "foldedComponentIds": Array [
                        "sc-bwzfXH",
                      ],
                      "render": [Function],
                      "styledComponentId": "sc-EHOje",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                  selected={false}
                  testId="existing-building"
                >
                  <Component
                    className="sc-bwzfXH sc-EHOje iRNIDo"
                    disabled={true}
                    onClick={[Function]}
                    selected={false}
                    testId="existing-building"
                  >
                    <button
                      className="sc-bwzfXH sc-EHOje iRNIDo"
                      data-testid="existing-building"
                      disabled={true}
                      onClick={[Function]}
                    >
                      ONBOARDING_EXISTING
                    </button>
                  </Component>
                </StyledComponent>
              </Styled(Styled(Component))>
              <Styled(Styled(Component))
                onClick={[Function]}
                selected={true}
                testId="new-building"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-EHOje",
                        "isStatic": false,
                        "lastClassName": "gATSaV",
                        "rules": Array [
                          "
  box-sizing: border-box;
  outline: none;
  border: ",
                          [Function],
                          ";
  border-radius: 2px;
  position: relative;
  background-color: ",
                          [Function],
                          ";
  min-width: 120px;
  color: #000000;
  opacity: ",
                          [Function],
                          ";
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
",
                          "
  text-align: center;
  padding: 28px 10px;
  margin-right: 20px;
  overflow: visible;
  :hover { ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "Styled(Styled(Component))",
                      "foldedComponentIds": Array [
                        "sc-bwzfXH",
                      ],
                      "render": [Function],
                      "styledComponentId": "sc-EHOje",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                  selected={true}
                  testId="new-building"
                >
                  <Component
                    className="sc-bwzfXH sc-EHOje gATSaV"
                    onClick={[Function]}
                    selected={true}
                    testId="new-building"
                  >
                    <button
                      className="sc-bwzfXH sc-EHOje gATSaV"
                      data-testid="new-building"
                      onClick={[Function]}
                    >
                      <Styled(CheckCircleIcon)
                        fontSize="default"
                        htmlColor="#6CB9D5"
                      >
                        <StyledComponent
                          fontSize="default"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "compare": null,
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bdVaJa",
                                "isStatic": false,
                                "lastClassName": "bYrkLc",
                                "rules": Array [
                                  "
  position: absolute;
  background-color: #ffffff;
  top: -11px;
  right: -11px;
  &.MuiSvgIcon-root {
    font-size: 1.5rem;
  }
",
                                ],
                              },
                              "displayName": "Styled(CheckCircleIcon)",
                              "foldedComponentIds": Array [],
                              "muiName": "SvgIcon",
                              "render": [Function],
                              "styledComponentId": "sc-bdVaJa",
                              "target": Object {
                                "$$typeof": Symbol(react.memo),
                                "compare": null,
                                "displayName": "CheckCircleIcon",
                                "muiName": "SvgIcon",
                                "type": Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "render": [Function],
                                },
                              },
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          htmlColor="#6CB9D5"
                        >
                          <ForwardRef
                            className="sc-bdVaJa bYrkLc"
                            fontSize="default"
                            htmlColor="#6CB9D5"
                          >
                            <WithStyles(ForwardRef(SvgIcon))
                              className="sc-bdVaJa bYrkLc"
                              fontSize="default"
                              htmlColor="#6CB9D5"
                            >
                              <ForwardRef(SvgIcon)
                                className="sc-bdVaJa bYrkLc"
                                classes={
                                  Object {
                                    "colorAction": "MuiSvgIcon-colorAction",
                                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                                    "colorError": "MuiSvgIcon-colorError",
                                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                    "root": "MuiSvgIcon-root",
                                  }
                                }
                                fontSize="default"
                                htmlColor="#6CB9D5"
                              >
                                <svg
                                  aria-hidden="true"
                                  className="MuiSvgIcon-root sc-bdVaJa bYrkLc"
                                  color="#6CB9D5"
                                  focusable="false"
                                  role="presentation"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                                  />
                                </svg>
                              </ForwardRef(SvgIcon)>
                            </WithStyles(ForwardRef(SvgIcon))>
                          </ForwardRef>
                        </StyledComponent>
                      </Styled(CheckCircleIcon)>
                      ONBOARDING_NEW_BUILDING
                    </button>
                  </Component>
                </StyledComponent>
              </Styled(Styled(Component))>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</NewExisting>
`;
